[RFC] Custom Scalar Specification URLs (#649)
graphql/graphql-spec: spec/Section 4 -- Introspection.md

  # should be non-null for NON_NULL and LIST only, must be null for the others
  ofType: __Type

>   # should be non-null for SCALAR only, must be null for the others

Reviewer: Nitpick: Currently in my draft for the reference implementation https://github.com/graphql/graphql-js/pull/2276, if you don't add the `@specified` directive to a scalar, `specifiedBy` will be `null`.

If that's the correct implementation, then perhaps this wording could be updated to something along the lines of:
```suggestion
  # should be a string or null for SCALAR only, must be null for the others
```

Author: Hmm, it's a good point but I think I'd prefer just replacing `should` with `can`.

>   specifiedBy: String

Reviewer 2: It correlates to some extent with #300 .
If this extension of the introspection schema will be added, it is logical to expect that the proposed in #300 extension 
 should be also accepted. Now there is a second server directive besides `@deprecated`, which is provided through introspection as a special property. I have nothing against this, but I believe that we need to provide a way to get user directives through introspection too. It will be a consistent step.

Reviewer 3: @sungam3r there are some experiments on introspection extensions. Technically you can already today extend introspection but this alway comes with the danger that you violate a future specification. 

@IvanGoncharov was experimenting on this one with GraphQL-js. At the moment you can put metadata on almost everything there. We do almost the same with hot chocolate. The next thing would be to allow to have something like extensions on introspection so that there is a point where you can cleanly extend the introspection types.

Doing this through directives brings some challenges since directives, specifically their arguments are input types that could bear challenges to automatically extend the introspection from those.

@IvanGoncharov maybe we should start a new discussion on that?

Reviewer 2: I have a separate graphql-dotnet branch in which directives are already returned via introspection. We use this forked preview version in production.

> this alway comes with the danger that you violate a future specification.

I agree. It’s not scary for us. We need new features now, and not when they (possibly) appear in the specification.

Reviewer 3: @sungam3r i did mean that more from them perspective. Hc allows extending the introspection already for a couple of versions. But the user that does that has to know that he/she might violate a future spec. We just have a warning in the docs that you can do it but may run into an issue in the future.

Represents scalar types such as Int, String, and Boolean. Scalars cannot have fields.

> A GraphQL type designer should provide use the `@specified` directive to provide

Reviewer 4: typo: `should use the`

a specification URL describing the data format, serialization, and coercion
rules for the scalar. It is recommended to also summarize the format and provide
an example in the description field.

Fields

* `kind` must return `__TypeKind.SCALAR`.
* `name` must return a String.

> * `specifiedBy` may return a URL or {null}.

Reviewer 5: We don't have a `URL` type so it should be `String`.

Author: maybe `may return a String (in the form of a URL) or {null}.` ?

Represents scalar types such as Int, String, and Boolean. Scalars cannot have fields.

> A GraphQL type designer should use the `@specifiedBy` directive to provide

Reviewer 6: ```suggestion
A GraphQL type designer should define `specifiedBy` for custom scalars as
```
